backend/local/backend.go

```diff
@@ -61,6 +79,10 @@ type Local struct {
 	schema *schema.Backend
 	opLock sync.Mutex
 	once   sync.Once
+
+	// workingDir is where the State* paths should be relative to.
+	// This is currently only used for tests.
+	workingDir string
```

Comment:
I'd avoid this and use `testChdir`. We actually USED to have a field like this but I've tried to make helpers to make sure we're executing code identical as much as possible (we're not perfect yet).